components/raftstore-v2/src/fsm/peer.rs

```diff
@@ -95,18 +113,103 @@ impl<EK: KvEngine, ER: RaftEngine> Fsm for PeerFsm<EK, ER> {
+    pub fn new(fsm: &'a mut PeerFsm<EK, ER>, store_ctx: &'a mut StoreContext<EK, ER, T>) -> Self {
         Self { fsm, store_ctx }
     }
 
-    pub fn handle_msgs(&self, peer_msgs_buf: &mut Vec<PeerMsg<EK>>) {
+    pub fn schedule_tick(&mut self, tick: PeerTick) {
+        assert!(PeerTick::VARIANT_COUNT <= u16::BITS as usize);
+        let idx = tick as usize;
+        let key = 1u16 << (idx as u16);
+        if self.fsm.tick_registry & key != 0 {
+            return;
+        }
+        if is_zero_duration(&self.store_ctx.tick_batch[idx].wait_duration) {
+            return;
+        }
+        trace!(
+            self.fsm.logger(),
+            "schedule tick";
+            "tick" => ?tick,
+            "timeout" => ?self.store_ctx.tick_batch[idx].wait_duration,
+        );
+
+        let region_id = self.fsm.peer.region_id();
+        let mb = match self.store_ctx.router.mailbox(region_id) {
+            Some(mb) => mb,
+            None => {
+                error!(
+                    self.fsm.logger(),
+                    "failed to get mailbox";
+                    "tick" => ?tick,
+                );
+                return;
+            }
+        };
+        self.fsm.tick_registry |= key;
+        let logger = self.fsm.logger().clone();
+        // TODO: perhaps following allocation can be removed.
+        let cb = Box::new(move || {
+            // This can happen only when the peer is about to be destroyed
+            // or the node is shutting down. So it's OK to not to clean up
+            // registry.
+            if let Err(e) = mb.force_send(PeerMsg::Tick(tick)) {
+                debug!(
+                    logger,
+                    "failed to schedule peer tick";
+                    "tick" => ?tick,
+                    "err" => %e,
+                );
+            }
+        });
+        self.store_ctx.tick_batch[idx].ticks.push(cb);
+    }
+
+    fn on_start(&mut self) {
+        self.register_raft_tick();
+    }
+
+    fn handle_tick(&mut self, tick: PeerTick) {
+        match tick {
+            PeerTick::Raft => self.on_raft_tick(),
+            PeerTick::RaftLogGc => unimplemented!(),
+            PeerTick::SplitRegionCheck => unimplemented!(),
+            PeerTick::PdHeartbeat => unimplemented!(),
+            PeerTick::CheckMerge => unimplemented!(),
+            PeerTick::CheckPeerStaleState => unimplemented!(),
+            PeerTick::EntryCacheEvict => unimplemented!(),
+            PeerTick::CheckLeaderLease => unimplemented!(),
+            PeerTick::ReactivateMemoryLock => unimplemented!(),
+            PeerTick::ReportBuckets => unimplemented!(),
+            PeerTick::CheckLongUncommitted => unimplemented!(),
+        }
+    }
+
+    pub fn handle_msgs(&mut self, peer_msgs_buf: &mut Vec<PeerMsg<EK>>) {
         for msg in peer_msgs_buf.drain(..) {
-            // TODO: handle the messages.
+            match msg {
```

Comment:
Also, I think it makes the code cleaner to unify how we handle different tick/msg. For example, in v1, handling a message usually goes to `on_xx_msg`, but there are exceptions (like propose cmd). How about we stick to `on_xx` this time, and group their definitions together as well.